test/pjrt/common.py

```diff
@@ -0,0 +1,92 @@
+import concurrent.futures
+import functools
+import os
+from typing import Any, Callable, Dict
+
+import torch_xla.core.xla_model as xm
+
+
+def pjrt_device() -> str:
+  return os.environ['PJRT_DEVICE']
+
+
+def num_visible_tpu_chips() -> int:
+  visible_devices = os.environ.get('TPU_VISIBLE_DEVICES')
+
+  return len(visible_devices.split(',')) if visible_devices else 4
+
+
+def configure_tpu_topology(rank: int, processes: int, base_port=8476):
+  '''Set default TPU topology environment variables for a single TPU host.'''
+  ports = list(range(base_port, base_port + processes))
+  os.environ.setdefault('TPU_CHIPS_PER_PROCESS_BOUNDS', '1,1,1')
+  os.environ.setdefault('TPU_PROCESS_BOUNDS', '2,2,1')
+  os.environ.setdefault('TPU_PROCESS_ADDRESSES',
+                        ','.join(f'localhost:{port}' for port in ports))
+
+  os.environ.setdefault('TPU_VISIBLE_DEVICES', str(rank))
+  os.environ.setdefault('TPU_PROCESS_PORT', str(ports[rank]))
```

Comment:
what is this `TPU_PROCESS_PORT` used for?

Reply:
It's an internal setting in libtpu that's only used when there are multiple communicating processes per TPU host. It corresponds to one of the ports in `TPU_PROCESS_ADDRESSES`